Lib/test/test_ftplib.py

```diff
@@ -1045,6 +1045,10 @@ def testTimeoutValue(self):
         self.evt.wait()
         ftp.close()
 
+        # bpo-39259
+        with self.assertRaises(ValueError):
+            ftplib.FTP(HOST, timeout=0)
```

Comment:
FTP_TLS is not tested, but I don't think that its a big deal. FTP_TLS requires to check if ssl is available, and the current test suite isn't properly designed to test FTP and FTP_TLS at the same time.